build-info-extractor/src/main/java/org/jfrog/build/extractor/Proxy.java

```diff
@@ -0,0 +1,81 @@
+package org.jfrog.build.extractor;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.net.InetSocketAddress;
+import java.net.ProxySelector;
+import java.net.URI;
+import java.util.List;
+
+import static java.net.Proxy.Type.HTTP;
+
+public class Proxy {
+
+    private final String host;
+    private final int port;
+    private final String username;
+    private final String password;
+    private final boolean https;
+
+    private Proxy(String host, int port, boolean https) {
+        this.host = host;
+        this.port = port;
+        this.https = https;
+        this.username = https ? System.getProperty("https.proxyUser") : System.getProperty("http.proxyUser");
+        this.password = https ? System.getProperty("https.proxyPassword") : System.getProperty("http.proxyPassword");
+    }
+
+    public static Proxy createFromSystemProperties(String repositoryUrl) {
+        if (isProxyPropertiesAvailable(true) && isProxyPropertiesAvailable(false)) {
+            return null;
+        }
+        if (!isValidRepoUrl(repositoryUrl)) {
+            return null;
+        }
+        List<java.net.Proxy> applicableProxies = ProxySelector.getDefault().select(URI.create(repositoryUrl));
+        java.net.Proxy proxy = applicableProxies.stream().filter((java.net.Proxy p) -> p.type() == HTTP).findFirst().orElse(null);
```

Comment:
Why only HTTP?
Let's prefer HTTPS and only if there is no matching HTTPS proxy use HTTP

Reply:
It represents the proxy type and not the actual protocols. Check out the Proxy description at `java.net`.